app/helpers/pipeline_runs_helper.rb

```diff
@@ -181,4 +181,25 @@ def pipeline_version_has_coverage_viz(pipeline_version)
   def pipeline_version_at_least_2(pipeline_version)
     pipeline_version_at_least(pipeline_version, PIPELINE_VERSION_2)
   end
+
+  def check_for_user_error(failed_stage)
+    if failed_stage.step_number == 1
+      user_input_validation_file = "#{failed_stage.pipeline_run.host_filter_output_s3_path}/#{PipelineRun::INPUT_VALIDATION_NAME}"
+      invalid_step_input_file = "#{failed_stage.pipeline_run.host_filter_output_s3_path}/#{PipelineRun::INVALID_STEP_NAME}"
+      if file_generated_since_run(failed_stage, user_input_validation_file)
+        file = Tempfile.new
+        downloaded = PipelineRun.download_file_with_retries(user_input_validation_file, file.path, 3, false)
+        validation_error = downloaded ? JSON.parse(File.read(file))["Validation error"] : nil
+        file.unlink
+        return ["FAULTY_INPUT", validation_error] if validation_error
+      end
+      if file_generated_since_run(failed_stage, invalid_step_input_file)
+        # Currently 'invalid_step_input_file' only gets produced in a single failure mode:
+        # INSUFFICIENT_READS. If we extended the file to other error types in the future,
+        # we would download 'invalid_step_input_file' and parse the failure mode here.
```

Comment:
why not future proof it now by downloading the file and parse it as above with `user_input_validation_file`?

and-or it would be good to double check that the output read count is actually zero for the previous pipeline step

Reply:
The idseq-dag change does the read counting: https://github.com/chanzuckerberg/idseq-dag/commit/861e4d9f1382315ae16971c6985b31f08feca501
No need to revalidate here in my opinion.
Yes, I will parse `user_input_validation_file` then :)

Reply:
ok done